### PR TITLE
Polish slappers (again)

### PR DESCRIPTION
**Pull request by StrawWagen**

### CODE REVIEW

File: lua/weapons/cfc_slappers/shared.lua [hunk 1]
```diff
@@ -159,7 +167,7 @@ function SWEP:ReactionForce( owner, tr, scale )
     local mul = GetConVar( "slappers_base_force" ):GetInt() * self:ForceMul() * scale
 
     local slapVel = -vec * mul
-    local vel = slapVel * 0.6 + origVel
```
Comment (wrefgtzweve) [inline lua/weapons/cfc_slappers/shared.lua:162]: magic number?

File: lua/weapons/cfc_slappers/shared.lua [hunk 2]
```diff
@@ -654,4 +681,4 @@ function SWEP:SecondaryAttack()
     self:Slap()
     self:SetNextSecondaryFire( CurTime() + self.Primary.Delay )
 
-end
+end
```
Comment (wrefgtzweve) [inline lua/weapons/cfc_slappers/shared.lua:684]: make sure to add a newline at the end of the file

File: lua/weapons/cfc_slappers/shared.lua [hunk 3]
```diff
@@ -140,6 +140,14 @@ function SWEP:MissEffect()
 
 end
 
+function SWEP:DamageMul()
```
Comment (wrefgtzweve) [inline lua/weapons/cfc_slappers/shared.lua:143]: this should be a swep variable, not a swep function, same for WeaponKnockWeight and ForceMul

File: lua/weapons/cfc_slappers/shared.lua [hunk 4]
```diff
@@ -140,6 +140,14 @@ function SWEP:MissEffect()
...
+    return 1
+end
+
+function SWEP:GetHull()
```
Comment (wrefgtzweve) [inline lua/weapons/cfc_slappers/shared.lua:147]: this is only used once, is it really needed?

File: lua/weapons/cfc_slappers/shared.lua (hunk 1 above)
Comment (StrawWagen) [reply lua/weapons/cfc_slappers/shared.lua:162]: Resolved

File: lua/weapons/cfc_slappers/shared.lua (hunk 2 above)
Comment (StrawWagen) [reply lua/weapons/cfc_slappers/shared.lua]: Applied

File: lua/weapons/cfc_slappers/shared.lua (hunk 3 above)
Comment (StrawWagen) [reply lua/weapons/cfc_slappers/shared.lua]: Applied!

File: lua/weapons/cfc_slappers/shared.lua (hunk 4 above)
Comment (StrawWagen) [reply lua/weapons/cfc_slappers/shared.lua]: Resolved!

File: lua/weapons/cfc_slappers/shared.lua [hunk 5]
```diff
@@ -568,23 +570,43 @@ function SWEP:Slap()
...
+    local traceDat = {
+        start = start,
+        endpos = endpos,
         filter = owner
-    } )
+    }
+
+    local tr = util.TraceLine( traceDat )
+
+    if not IsValid( tr.Entity ) and world ~= tr.Entity then
```
Comment (wrefgtzweve) [inline lua/weapons/cfc_slappers/shared.lua:591]: why are you doing two traces here?

Comment (StrawWagen) [reply lua/weapons/cfc_slappers/shared.lua:591]: -- always hit what is aimed at, and if there's nothing there, then do the hull
so if someone is aiming at a player in a tight space, hit the player, not the walls around them

**Review by wrefgtzweve — APPROVED** [review]

**Pull request merged**